core/ffi/types/build.rs

```diff
@@ -1,61 +1,37 @@
 // Copyright (c) 2022 The MobileCoin Foundation
 //! Builds the FFI type bindings for the common SGX SDK types
 
-use bindgen::{callbacks::ParseCallbacks, Builder};
-
-// Types that don't have an SGX qualifier.
-//
-// These types are of the form <name> and later typedefed to an `sgx_<name>`.
-//
-// ```C
-//      typedef struct _foo_name {
-//          int a;
-//          float b;
-//      } sgx_foo_name;
-// ```
-//
-// To keep the noise out of the bindings, we use the underlying type and tell
-// bindgen to map directly to `sgx_<name>` version.
-const ALLOWED_UNDERLYING_TYPES: &[&str] = &[
-    "_status_t",
-    "_target_info_t",
-    "_attributes_t",
-    "_report_t",
-    "_key_request_t",
-];
-
-#[derive(Debug)]
-struct Callbacks;
-
-impl ParseCallbacks for Callbacks {
-    fn item_name(&self, name: &str) -> Option<String> {
-        if name.starts_with("_sgx") {
-            Some(name[1..].to_owned())
-        } else if name.starts_with('_') {
-            Some(format!("sgx{}", name))
-        } else {
-            None
-        }
-    }
-}
+use mc_sgx_core_build::SGXParseCallbacks;
 
 fn main() {
     let sgx_library_path = mc_sgx_core_build::sgx_library_path();
-    let mut builder = Builder::default()
+    let builder = mc_sgx_core_build::sgx_builder()
         .header_contents(
             "core_types.h",
             "#include <sgx_error.h>\n#include <sgx_report.h>",
         )
         .clang_arg(&format!("-I{}/include", sgx_library_path))
         .newtype_enum("_status_t")
-        .parse_callbacks(Box::new(Callbacks))
-        .ctypes_prefix("core::ffi")
-        .use_core()
-        .allowlist_type("sgx_key_128bit_t");
-
-    for t in ALLOWED_UNDERLYING_TYPES.iter() {
-        builder = builder.allowlist_type(t)
-    }
+        .parse_callbacks(Box::new(SGXParseCallbacks))
+        .allowlist_type("sgx_key_128bit_t")
```

Comment:
What's the reasoning behind adding an explicit `allowlist_type` for each type instead of keeping the for loop? The for loop makes it slightly easier to add a new type becasue you don't need to type out `allowlist_type` again, but I guess this way is slightly more readable?

Reply:
I'm fine reverting, not really invested one way or the other.
 did this as I chewed on this comment, https://github.com/mobilecoinfoundation/sgx/pull/9#discussion_r927103062

I think my thought was to mimic the other crates that are being more explicit about the types, but perhaps they would be better served having a static slice and we make a common utility to add a slice of allowed types

Reply:
SGTM.

Reply:
Thinking about it more, I agree the const slice is nicer, I also sorted them, should have done that earlier sorry, but now I know clion's vim bindings support `sort`